### PR TITLE
Missing features for invoice

### PR DESCRIPTION
**Pull request by woller**
This PR adds two missing features from invoice: the ability to supply filter options to the invoice end point and a model for line items on an invoice.

### CODE REVIEW

File: spec/harvesting/harvest_data_setup.rb [hunk 1]
```diff
@@ -337,4 +337,6 @@ def cassette_let!(name, &block)
     result
   end
 
+  let!(:invoices) do
+  end
```
Comment (etagwerker) [inline spec/harvesting/harvest_data_setup.rb:341]: @woller Why is this necessary?

Comment (woller) [reply spec/harvesting/harvest_data_setup.rb]: It is not. I added it, when trying to figure out how to get vcr working and forgot to remove it.
